test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -969,6 +969,14 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
             $response.Content.Headers."Authorization" | Should -BeExactly "test"
         }
 
+        It "Validates Invoke-WebRequest with -WebSession, -PreserveAuthorizationOnRedirect and -MaximumRedirection doesn't change session.MaximumRedirection on multiple redirects: <redirectType>" -TestCases $redirectTests {
+            param($redirectType)
+            $uri = Get-WebListenerUrl -Test 'Redirect' -TestValue 2 -Query @{type = $redirectType}
+            $null = Invoke-WebRequest -Uri $uri -PreserveAuthorizationOnRedirect -MaximumRedirection 2 -SessionVariable session
+
+            $session.MaximumRedirection | Should -BeExactly 2
```

Comment:
Hmm, the problem could be with any another parameter. Maybe add tests for them too.

Reply:
I will add the tests in another PR

Reply:
We currently don't have this problem with other parameters, because only session.MaximumRedirection is modified in the handle redirect loop `WebSession.MaximumRedirection--`.
Should I add the tests anyway?

Reply:
Added all the tests

Reply:
Why are you adding unrelated tests to this PR? Let's keep just the original test

Reply:
Or set correct name for test.
And why you creates session object with cmdlet if you can create it with constructor that faster.

Reply:
Reworked the tests